userland/Makefile

```diff
@@ -31,6 +31,9 @@ CPPFLAGS += \
 	    -mpic-register=r9\
 	    -mno-pic-data-is-text-relative
 
+$(BUILDDIR)/%.o: %.c | $(BUILDDIR)
+	$(CC) $(CFLAGS) $(CPPFLAGS) -c -o $@ $<
+
```

Comment:
This was actually intentionally separate. The idea is to separate the responsibilities of the common Makefile and the app specific Makefile.

The app specific Makefile figures out where source files are and how to compile them to object files (e.g. using `gcc` from the main directory, but potentially with another compiler or from other source folders as the case with the ble libraries).

The common Makefile basically plays the role of a platform-specific linker: it's only input is `OBJS` variable (containing object files to be linked), pulls in the right libc and links everything together.